frontend/src/metabase/setup/actions.ts

```diff
@@ -197,6 +203,13 @@ export const submitSetup = createAsyncThunk<void, void, ThunkConfig>(
         },
       });
 
+      if (usageReason === "embedding" || usageReason === "both") {
+        setShowEmbedHomepageFlag();
+      } else {
+        // make sure that state is clean in case of more than one setup on the same browser
```

Comment:
Vitorio wasn't able to test the "normal" cases after he triggered the homepage, because he probably didn't dismissed it and the flag stayed in `localStorage` in the following setups.
Multiple setups on the same browser is a pretty rare edge case but the fix is just cleaning the state